index.d.ts

```diff
@@ -0,0 +1,226 @@
+        once(event: 'move', listener: (file: MutableFile, oldDir: MutableFile) => void): this;
+        once(event: 'ready', listener: (storage: this) => void): this;
+        once(event: 'update', listener: (file: MutableFile) => void): this;
+        once(event: 'delete', listener: (file: Readonly<File>) => void): this;
+    }
+    export class API extends EventEmitter {
+        fetch: Fetch;
+        gateway: string;
+        counterId: string;
+        userAgent: string;
+        keepalive: boolean;
+        httpAgent: HttpAgent;
+        httpsAgent: HttpsAgent;
+        sn?: AbortController;
+        static globalApi?: API;
+        static getGlobalApi(): API;
+        constructor(keepalive: boolean, opts?: APIOpts);
+        close(): void;
+        pull(sn: AbortController, retryno?: number): void;
+        wait(url: fetch.RequestInfo, sn: AbortController): void;
+        defaultFetch(url: fetch.RequestInfo, opts?: fetch.RequestInit): Fetch;
+        request(json: Object, cb: (error: err, response?: any) => void, retryno?: number): void;
+    }
+
+    export class File extends EventEmitter {
+        api: API;
+        type: number;
+        size?: number;
+        label: string;
+        owner?: string;
+        nodeId?: string;
+        loadedFile?: string; // not entirely sure
+        downloadId: string;
+        directory: boolean;
+        favorited: boolean;
+        timestamp?: number;
+        key: Nullable<Buffer>;
+        name: Nullable<string>;
+        attributes: BufferString;
+        constructor(opts: FileOpts);
+        static fromURL(opt: FileOpts | string, extraOpt?: Partial<FileOpts>): File;
+        static unpackAttributes(at: Buffer): void | Object;
+        static defaultHandleRetries(tries: number, error: err, cb: errorCb): void;
+        get createdAt(): number;
+        loadAttributes(cb: BufferString): this;
+        parseAttributes(at: BufferString): void;
+        decryptAttributes(at: BufferString): this;
+        loadMetadata(aes: AES, opt: metaOpts): void;
+        checkConstructorArgument(value: BufferString): void;
+        link(options: linkOpts | boolean, cb: (error: err, url: string) => void): void;
+        download(options: downloadOpts, cb?: (error: err, data: Buffer) => void): Readable;
+    }
+    export class MutableFile extends File {
+        storage: Storage;
+        static packAttributes(attributes: Object): Buffer;
+        constructor(opts: FileOpts, storage: Storage);
+        mkdir(opts: mkdirOpts | string, cb?: (error: err, file: Nullable<MutableFile>) => void): void;
+        upload(opts: uploadOpts | string, source?: BufferString, cb?: uploadCb): Writable;
+        uploadAttribute(type: 0 | 1, data: Buffer, callback?: (error: err, file?: this) => void): void;
+        // Not sure about type of file in delete's cb
+        delete(permanent?: boolean, cb?: (error: err, file?: File) => void): this;
+        moveTo(target: File | string, cb?: (error: err, file?: File) => void): this;
```

Comment:
I'm not sure what the second parameter of the callback is so I made a guess, could be wrong